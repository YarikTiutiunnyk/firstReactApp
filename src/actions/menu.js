export const setActivePage = (string)=>({
    type: "SET_ACTIVE_PAGE",
    payload: string
})