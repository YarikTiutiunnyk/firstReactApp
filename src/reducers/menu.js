const initiaState = {
    activePage: 'collections'
}

export default (state = initiaState, action)=>{
    switch (action.type) {
        case 'SET_ACTIVE_PAGE':   
            return {
                ...state,
                activePage:  action.payload
            };        
        
        default:
            return state;
    }
}