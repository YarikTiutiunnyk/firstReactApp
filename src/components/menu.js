import React, { Component } from 'react'
import { connect } from "react-redux";
import { Link } from 'react-router-dom';
import { Menu } from 'semantic-ui-react'

import { setActivePage } from '../actions/menu';

class MenuExampleBasic extends Component {
  constructor(props) {
    super(props);
    this.state = {activeItem: 'collections'};

    this.handleItemClick = this.handleItemClick.bind(this);
  }

  //state = {activeItem: 'collections'}


  handleItemClick = (e, { name }) => {
    this.setState({ activeItem: name })
    this.props.setActivePage(name);
  }
  render() {
    const { activeItem } = this.state

    return (
      <Menu widths={3} stackable={true}>
        <Menu.Item
          as={Link} to='/'
          name='collections'
          active={activeItem === 'collections'}
          onClick={this.handleItemClick}
        >
          My collectoons
        </Menu.Item>

        <Menu.Item
          as={Link} to='/films'
          name='films'
          active={activeItem === 'films'}
          onClick={this.handleItemClick}
        >
          Serch films
        </Menu.Item>    

        <Menu.Item
          as={Link} to='/series'
          name='series'
          active={activeItem === 'series'}
          onClick={this.handleItemClick}
        >
          Serch series
        </Menu.Item>
      </Menu>
    )
  }
}

//Функція зв'язує redux state з props 
const mapStateToProps = state => {
  const { activePage } = state.menu;
  return { activePage};
};

export default connect(
  mapStateToProps,
  { setActivePage }
)(MenuExampleBasic);