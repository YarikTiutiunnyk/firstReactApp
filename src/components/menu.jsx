import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Menu } from 'semantic-ui-react';

import { setActivePage } from '../actions/menu';

export class MenuExampleBasic extends Component {
  handleItemClick = (name11) => {
    this.props.setActivePage(name11);
  };

  render() {
    const activeItem = this.props.activePage;

    return (
      <Menu widths={3} stackable={true}>
        <Menu.Item
          as={Link}
          to="/firstReactApp/"
          name="collections"
          active={activeItem === 'collections'}
          onClick={this.handleItemClick.bind(this, 'collections')}
        >
          My collectoons
        </Menu.Item>

        <Menu.Item
          as={Link}
          to="/firstReactApp/films"
          name="films"
          active={activeItem === 'films'}
          onClick={this.handleItemClick.bind(this, 'films')}
        >
          Serch films
        </Menu.Item>

        <Menu.Item
          as={Link}
          to="/firstReactApp/series"
          name="series"
          active={activeItem === 'series'}
          onClick={this.handleItemClick.bind(this, 'series')}
        >
          Serch series
        </Menu.Item>
      </Menu>
    );
  }
}
MenuExampleBasic.propTypes = {
  setActivePage: PropTypes.func.isRequired,
  activePage: PropTypes.string.isRequired
};

//Функція зв'язує redux state з props
const mapStateToProps = (state) => {
  const { activePage } = state.menu;
  return { activePage };
};

export default connect(mapStateToProps, { setActivePage })(MenuExampleBasic);
