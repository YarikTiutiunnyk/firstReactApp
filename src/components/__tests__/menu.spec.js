import React from 'react';
import { shallow } from 'enzyme';
import { MenuExampleBasic as Menu } from '../menu';
//import configureStore from 'redux-mock-store';

describe('Menu test', () => {
  const mocksetActivePage = jest.fn();
  const props = {
    activePage: 'collections',
    setActivePage: mocksetActivePage
  };

  describe('Menu initial', () => {
    const wrapper = shallow(<Menu {...props} />);
    //console.log(wrapper.find('MenuItem[name="collections"]').debug());

    it('Collections actibve renders properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Menu events', () => {
    const wrapper = shallow(<Menu {...props} />);

    it('Click on films triggered the function setActivePage(films)', () => {
      wrapper.find('MenuItem[name="films"]').simulate('click');
      expect(mocksetActivePage.mock.calls[0][0]).toEqual('films');
    });

    it('Click on series triggered the function setActivePage(series)', () => {
      wrapper.find('MenuItem[name="series"]').simulate('click');
      expect(mocksetActivePage.mock.calls[1][0]).toEqual('series');
    });

    it('Click on collections triggered the function setActivePage(collections)', () => {
      wrapper.find('MenuItem[name="collections"]').simulate('click');
      expect(mocksetActivePage.mock.calls[2][0]).toEqual('collections');
    });
  });
});
