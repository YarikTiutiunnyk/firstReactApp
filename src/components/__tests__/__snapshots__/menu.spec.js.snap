// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu test Menu initial Collections actibve renders properly 1`] = `
<Menu
  stackable={true}
  widths={3}
>
  <MenuItem
    active={true}
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    name="collections"
    onClick={[Function]}
    to="/firstReactApp/"
  >
    My collectoons
  </MenuItem>
  <MenuItem
    active={false}
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    name="films"
    onClick={[Function]}
    to="/firstReactApp/films"
  >
    Serch films
  </MenuItem>
  <MenuItem
    active={false}
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    name="series"
    onClick={[Function]}
    to="/firstReactApp/series"
  >
    Serch series
  </MenuItem>
</Menu>
`;
